pkgs/by-name/ar/arrpc/package.nix

```diff
@@ -1,31 +1,39 @@
-{ lib
-, buildNpmPackage
-, fetchFromGitHub
-}: buildNpmPackage rec {
+{
+  lib,
+  buildNpmPackage,
+  fetchFromGitHub,
+}:
+buildNpmPackage rec {
   pname = "arrpc";
-  version = "3.4.0";
+  version = "3.5.0";
 
   src = fetchFromGitHub {
     owner = "OpenAsar";
     repo = "arrpc";
-    # Release commits are not tagged
-    # release: 3.4.0
-    rev = "cca93db585dedf8acc1423f5e2db215de95c4c3b";
-    hash = "sha256-SeegrCgbjfVxG/9xfOcdfbVdDssZOhjBRnDipu6L7Wg=";
+    rev = version;
+    hash = "sha256-3xkqWcLhmSIH6Al2SvM9qBpdcLzEqUmUCgwYBPAgVpo=";
   };
 
-  npmDepsHash = "sha256-S9cIyTXqCp8++Yj3VjBbcStOjzjgd0Cq7KL7NNzZFpY=";
+  npmDepsHash = "sha256-lw6pngFC2Pnk+I8818TOTwN4r+8IsjvdMYIyTsTi49g=";
 
   dontNpmBuild = true;
 
+  postInstall = ''
+    mkdir -p $out/share/systemd/user
+    substitute ${./arrpc.service} $out/share/systemd/user/arrpc.service \
+      --subst-var-by arrpc $out/bin/arrpc
```

Comment:
I'm not sure if this is standard practice, would be the first time I see this in nixpkgs.

Reply:
I think it is standard. Packages like `thunar` and `syncthing` all installs a service file there. Example:

https://github.com/NixOS/nixpkgs/blob/c08e797454167d72e105c984a2e7452f15a99c07/pkgs/desktops/xfce/core/thunar/wrapper.nix#L22-L31

It will be picked up by system if `config.systemd.packages` includes the package. My concern is whether it should be a system unit or a user unit.

Reply:
The example you've provided is for vendoring a service provided by upstream - not for shipping our own, though I can't exactly see an issue with it...

I am in favor of having it as an user service, since that's already what the [home-manager module does](https://github.com/nix-community/home-manager/blob/master/modules/services/arrpc.nix)

Reply:
Alright it seems that I am wrong. I need to define a nixos module for that.

Reply:
https://discourse.nixos.org/t/how-to-enable-upstream-systemd-user-services-declaratively/7649

Honestly I still don't quite understand how systemd user services provided by packages work in NixOS. There are some discussions.